### PR TITLE
Feature/use selenide for prime ui

### PR DESCRIPTION
**Pull request by ivy-lli**
Refactor PrimeUi Tester:
- Separate in different class files
- Use Selenide as Testframework
- Raise version to 1.0.0-SNAPSHOT, update README

### CODE REVIEW

**Comment by ivy-lli** [comment]
Build: https://jenkins.ivyteam.io/job/supplements-primeui-tester/job/feature%252FuseSelenideForPrimeUi/

File: README.md [hunk 1]
```diff
@@ -1,18 +1,68 @@
...
-    	  <artifactId>primeui-tester</artifactId>
-    	  <version>0.0.5-SNAPSHOT</version>
-        <scope>tests</scope>
-      </dependency>
-    </dependencies>
+```xml
+<dependencies>
+...
+  <dependency>
+    <groupId>com.codeborne</groupId>
```
Comment (ivy-rew) [inline README.md]: can make the required deps smaller?
improve dependency pain by declaring optional/test scoped deps in primeui-tester?
shouldn't some deps of prime-ui be optional as well?

File: primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Accordion.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+
+import com.axonivy.ivy.supplements.primeui.tester.PrimeUi;
+import com.codeborne.selenide.SelenideElement;
+import com.codeborne.selenide.WebDriverRunner;
+
+public class Accordion
+{
+  private String accordionId;
```
Comment (ivy-rew) [inline primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Accordion.java]: final

File: primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Accordion.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    accorionTab(tabName).click();
+    String tabContentId = StringUtils.removeEnd(accorionTab(tabName).getAttribute("id"), "_header");
+
+    //Selenide doesn't support if a style is set or not. Primefaces 7 needs to wait for full visibility (no opacity).
+    PrimeUi.ajax(WebDriverRunner.getWebDriver()).await(ExpectedConditions.numberOfElementsToBe(
+            By.xpath("//div[@id='" + tabContentId + "' and not(contains(@style,'overflow'))]"), 1));
+    $(By.id(tabContentId)).shouldBe(attribute("aria-hidden", previousState));
+  }
+
+  private SelenideElement accorionTab(String tabName)
```
Comment (ivy-rew) [inline primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Accordion.java]: typo: accordion

File: primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Dialog.java [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+import static com.codeborne.selenide.Condition.exist;
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+
+import org.openqa.selenium.By;
+
+public class Dialog
+{
+  private String dialogId;
```
Comment (ivy-rew) [inline primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/Dialog.java]: final

File: primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/SelectCheckboxMenu.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+  public void selectAllItems()
+  {
+    openCheckboxPanel();
+    $(By.id(checkBoxMenuId + "_panel")).find(".ui-widget-header .ui-chkbox-box").shouldBe(visible).click();
+    $(By.id(checkBoxMenuId + "_panel")).find(".ui-widget-header .ui-chkbox-box").shouldHave(cssClass("ui-state-active"));
+    closeCheckboxPanel();
+  }
+
+  public void selectItemByValue(String labelValue)
```
Comment (ivy-rew) [inline primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/SelectCheckboxMenu.java]: should delegate impl to selectItemsByValue()

File: primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/SelectOneRadio.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    $(radioById(id)).shouldHave(cssClass("ui-state-active"));
+  }
+
+  public void selectItemByValue(final String value)
+  {
+    $(radioByLabel(value)).click();
+    $(radioByLabel(value)).shouldHave(cssClass("ui-state-active"));
+  }
+  
+  public String getSelected()
```
Comment (ivy-rew) [inline primeui-tester/src/main/java/com/axonivy/ivy/supplements/primeui/tester/widget/SelectOneRadio.java]: getSelectedValue()  ....?

File: primeui-tester/src/test/java/com/axonivy/ivy/supplements/primeui/tester/TestPrimeUi.java [hunk 7]
```diff
@@ -1,166 +1,119 @@
...
-            + "Brasilia");
-  }
-
-  private SelectCheckboxMenu getMultipleCheckboxMenu()
+  public void testSelectCheckBoxMenu_itemsByValue()
   {
-    WebElement checkbox = getElementWithLabel("Multiple:");
-    return prime.selectCheckboxMenu(By.id(checkbox.getAttribute("id")));
+    open("https://primefaces.org/showcase/ui/input/checkboxMenu.xhtml");
+    PrimeUi.selectCheckboxMenu(selectMenuForLabel("Multiple:")).selectItemsByValue("Miami", "Brasilia");
```
Comment (ivy-rew) [inline primeui-tester/src/test/java/com/axonivy/ivy/supplements/primeui/tester/TestPrimeUi.java]: I'd prefer static imports for 'PrimeUi' to make tests even more expressive.

File: primeui-tester/src/test/java/com/axonivy/ivy/supplements/primeui/tester/TestPrimeUi.java [hunk 8]
```diff
@@ -1,166 +1,119 @@
...
-
   @Test
-  public void testSelectCheckBoxMenu_all() throws Exception
+  public void testSelectCheckBoxMenu_all()
   {
-    driver.get("http://primefaces.org/showcase/ui/input/checkboxMenu.xhtml");
-    
-    getCheckboxMenu().selectAllItems();
-    submitAndCheck("Brasilia");
+    open("https://primefaces.org/showcase/ui/input/checkboxMenu.xhtml");
```
Comment (ivy-rew) [inline primeui-tester/src/test/java/com/axonivy/ivy/supplements/primeui/tester/TestPrimeUi.java]: this class has two highly duplicated parts:
- url to start with in showcase ( may isolate in a navigator class, but at least introduce a common uri)
- firstItem selection (isolate in a nice utility: maybe combine with navigator) ... e.g.      SelectCheckBoxMenu menu = showcase(inputCHeckbox).first().checkbox();
...assertStuff();

**Pull request merged**